src/libraries/System.Text.RegularExpressions/tests/RegexParserTests.netcoreapp.cs

```diff
@@ -0,0 +1,148 @@
+        [InlineData(@"(?<cat>cat)\w+(?<dog-16>dog)", RegexOptions.None, RegexParseError.UndefinedNumberedReference, 23)]
+        [InlineData(@"cat(?<->dog)", RegexOptions.None, RegexParseError.CaptureGroupNameInvalid, 7)]
+        [InlineData("a{2147483648}", RegexOptions.None, RegexParseError.QuantifierOrCaptureGroupOutOfRange, 12)]
+        [InlineData("a{2147483648,}", RegexOptions.None, RegexParseError.QuantifierOrCaptureGroupOutOfRange, 12)]
+        [InlineData("a{0,2147483647}", RegexOptions.None, null)]
+        [InlineData("a{0,2147483648}", RegexOptions.None, RegexParseError.QuantifierOrCaptureGroupOutOfRange, 14)]
+        // Surrogate pair which is parsed as [char,char-char,char] as we operate on UTF-16 code units.
+        [InlineData("[\uD82F\uDCA0-\uD82F\uDCA3]", RegexOptions.IgnoreCase, RegexParseError.ReversedCharacterRange, 5)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void Parse_NotNetFramework(string pattern, RegexOptions options, object error, int offset = -1)
+        {
+            Parse(pattern, options, error, offset);
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsBinaryFormatterSupported))]
+        public void RegexParseException_Serializes()
+        {
+#pragma warning disable RE0001 // Regex issue: Not enough )'s
+            ArgumentException e = Assert.ThrowsAny<ArgumentException>(() => new Regex("(abc|def"));
+#pragma warning restore RE0001 // Regex issue: Not enough )'s
+
+            var bf = new BinaryFormatter();
+            var s = new MemoryStream();
+            bf.Serialize(s, e);
+            s.Position = 0;
+
+            ArgumentException e2 = (ArgumentException)bf.Deserialize(s);
+            Assert.Equal(e.Message, e2.Message);
+        }
+
+        /// <summary>
+        /// Checks if action throws either a RegexParseException or an ArgumentException depending on the
+        /// environment and the supplied error.
+        /// </summary>
+        /// <param name="error">The expected parse error</param>
+        /// <param name="action">The action to invoke.</param>
+        static partial void Throws(RegexParseError error, int offset, Action action)
+        {
+            try
+            {
+                action();
+            }
+            catch (RegexParseException e)
+            {
+                RegexParseError regexParseError = e.Error;
+
+                // Success if provided error matches.
+                if (error == regexParseError)
+                {
+                    Assert.Equal(offset, e.Offset);
+                    return;
+                }
+
+                throw new XunitException($"Expected RegexParseException with error: ({error}) -> Actual error: {regexParseError})");
+            }
+            catch(Exception e)
+            { 
+                throw new XunitException($"Expected RegexParseException -> Actual: ({e.GetType()})");
```

Comment:
This shouldn't include `error` and `offset`?

Reply:
And we're only including the type of the exception?
```suggestion
                throw new XunitException($"Expected RegexParseException -> Actual: {e}");
```

Reply:
I don't understand the question, we test here for offset, type and parse error enum, the same method for netfx doesn't, because that info isn't there. Printing the type on error was original, but we can print the whole error, np.

Reply:
Including the data passed on helps debugging, and not including the whole exception message and stack hampers it.